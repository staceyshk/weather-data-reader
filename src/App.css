.App {
  text-align: left;
  background-color: #282c34;
  min-height: 100vh;
  color: #fff;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.csv-upload-form {
  margin: 40px 20px;
}

.csv-form-upload {
  margin: 10px;
}

.csv-upload-btn {
  margin-left: 10px;
}

.output {
  margin: 20px;
  font-weight: bold;
}

.result {
  color: green;
}

.error {
  color: red;
}
