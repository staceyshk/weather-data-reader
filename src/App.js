import { useState } from 'react'
import { getWeatherDataFromCSV, lowestSpreadDay } from './Utils'
import './App.css';

function App() {
  const [error, setError] = useState('')
  const [lowestDayText, setLowestDayText] = useState('')

  const handleSubmit = (event) => {
    event.preventDefault()
    const csv = event.target[0].files[0]

    if (!csv) {
      setError('Please upload a csv')
      setTimeout(600, setError(''))
    }

    const reader = new FileReader();

    reader.onload = function (e) {
      const text = e.target.result;
      try {
        const weatherData = getWeatherDataFromCSV(text)
        const lowestDays = lowestSpreadDay(weatherData)
        setLowestDayText(lowestDays.join(' '))
      } catch (error) {
        setError(error)
        setTimeout(600, setError(''))
      }
    }
  
    reader.readAsText(csv)
  }

  return (
    <div className="App">
      <header className="App-header">
        <p>Weather Data Spread Processor</p>
      </header>
      <section>
        <form className="csv-upload-form" onSubmit={handleSubmit}>
          <label htmlFor="csvFile">Upload CSV</label>
          <input className="csv-upload-btn" type="file" id="csvFile" accept=".csv" />
          <br />
          <br />
          <input type="submit" value="Process Weather Data" />
        </form>
        <section className="output">
          {error && <p className="error">{error}</p>}
          {lowestDayText && <p className="result">The lowest day(s) in this set are: {lowestDayText}</p>}
        </section>
      </section>
    </div>
  )
}

export default App
