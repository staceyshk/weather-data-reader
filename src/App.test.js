import { render, screen } from '@testing-library/react';
import App from './App';

// I ran out of time (work had to happen) to do further ui testing
test('renders learn react link', () => {
  const tree = render(<App />);
  expect(tree).toMatchSnapshot();
});
