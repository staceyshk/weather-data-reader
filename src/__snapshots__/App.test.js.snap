// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <p>
            Weather Data Spread Processor
          </p>
        </header>
        <section>
          <form
            class="csv-upload-form"
          >
            <label
              for="csvFile"
            >
              Upload CSV
            </label>
            <input
              accept=".csv"
              class="csv-upload-btn"
              id="csvFile"
              type="file"
            />
            <br />
            <br />
            <input
              type="submit"
              value="Process Weather Data"
            />
          </form>
          <section
            class="output"
          >
            
            
          </section>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <p>
          Weather Data Spread Processor
        </p>
      </header>
      <section>
        <form
          class="csv-upload-form"
        >
          <label
            for="csvFile"
          >
            Upload CSV
          </label>
          <input
            accept=".csv"
            class="csv-upload-btn"
            id="csvFile"
            type="file"
          />
          <br />
          <br />
          <input
            type="submit"
            value="Process Weather Data"
          />
        </form>
        <section
          class="output"
        >
          
          
        </section>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
